community-engagement-plan.md

```diff
@@ -0,0 +1,210 @@
+
+### Project name
+
+Internally, the project is called "IDMOCP".  We have to decide if
+this name is suitable for the public face of the project, or whether
+something else, like *FreeIPA on OpenShift* is better.
+
+**DECIDE**
+
+- IDMOCP?
+- Something else? (please share your ideas)
+
+
+### Code of Conduct
+
+The project should have a code of conduct, and we should make it
+clear what that is, and violation will be dealt with.  We may be
+able to operate under the FreeIPA code of conduct.
+
+**NOTE** I have follow up with FreeIPA project about operating under
+their CoC and reporting/enforcement mechanisms.
+
+### Communication channels
+
+#### IRC
+
+FreeIPA has an established IRC channel (`#freeipa` on
+`libera.chat`), that receives a fair bit of community engagement.
+
+**DECIDE** do we:
+
+- use `#freeipa` for IDMOCP discussion?  (We should consult FreeIPA
+  project about this option)
+- create our own IRC channel, e.g. `#freeipa-openshift`
+- not use IRC as a primary channel, and redirect questions in 
+  `#freeipa` to another venue.
+
+
+#### Slack
+
+Slack seems popular in the OpenShift and Kubernetes communities.  We
+could create a Slack channel specifically for the IDMOCP project.
+Slack has the advantage of keeping a permanent, searchable record of
+messages exchanged, and is a more expressive medium than IRC (links,
+attachments, images, etc).
+
+**DECIDE**
+
+- yes to slack
+- no to slack, but yes to something similar
+- no to slack or similar 
+
+#### Mailing list
+
+**DECIDE**
+
+In light of other communication channels (including GitHub issues or
+other ticket systems), do we require a public mailing list for the
+IDMOCP effort?
+
+- yes to mailing list
+- no to mailing list
+
+
+### Web resources
+
+Establish a web "front page" for this project, which must:
+
+- Outline the project goals and use cases, linking to pages or
+  resources with more detail
+
+- Point to the offical communication channels
+
+- Mention and link to the Code of Conduct
+
+- Contain (or link to) a contributer guide, which must include a
+  comprehensive list of code repositories, container image repos,
+  documentation, etc.
+
+- Link to recent and/or important blog posts, talks, demos, that
+  give people another way to understand what the project is about.
+
+**DECIDE** where to host this:
+
+- Page(s) on the main FreeIPA site, with in-links from repos and the
+  FreeIPA front page
+
+- Host the site on github, e.g.
+  `https://freeipa.github.io/freeipa-openshift`.
+
+- Somewhere else?
+
+### Code, issues and pull requests
+
+- All team code repositories should be public.  Repositories that
+  are currently private should be carefully checked for secrets in
+  the history, scrubbing them before making them public.
+
+- Create issue and pull request templates to assist contributors to
+  more easily provide the information we want to receive.
+
+### Building awareness and engagement
+
+- **Blog posts:** team members are encouraged to blog about their
```

Comment:
new RH blog-based project: https://developers.redhat.com/

Reply:
Thanks, I've incorporated this into the document.